tests/apollo/util/bft_debug_tool.py

```diff
@@ -0,0 +1,144 @@
+            name = 'UBSAN'
+        if 'TSAN_OPTIONS' in os.environ:
+            assert name == None, f"name={name} is already set! You should only run with a single debug tool!"
+            name = 'TSAN'
+        if 'ASAN_OPTIONS' in os.environ:
+            assert name == None, f"name={name} is already set! You should only run with a single debug tool!"
+            name = 'ASAN'
+        if os.environ.get('ENABLE_HEAPTRACK', default="false").lower() == "true":
+            assert name == None, f"name={name} is already set! You should only run with a single debug tool!"
+            name = "HEAPTRACK"
+        if not name:
+            # not running any debug tool, make it explicit
+            self.name = None
+            return
+
+        self.name = name
+
+        #
+        #  currently, do not run other tools at the same time, and we must run with graceful shutdown (SIGTERM sent to tester replica)
+        #
+        # Do not allow running debug tool without graceful shutdown
+        assert os.environ.get('GRACEFUL_SHUTDOWN', "").lower() in ["true", "on"], \
+            f"{name} should be ran with graceful shutdown only!"
+        # Do not allow running debug tool with PERF_REPLICA enabled
+        assert not os.environ.get('PERF_REPLICA', ""), \
+            f"{name} should be ran without PERF_REPLICA defined!"
+        # Do not allow running debug tool with CODECOVERAGE enabled
+        assert not os.environ.get('CODECOVERAGE', "").lower() in ["true", "on"], \
+            f"{name} should be ran without CODECOVERAGE defined!"
+
+        output_folder_name = name.lower() + "_logs"
+        test_suite_name = os.environ.get('TEST_NAME')
+        if os.environ.get('BLOCKCHAIN_VERSION', default="1").lower() == "4" :
+            test_suite_name += "_v4"
+        self.output_log_path = os.path.join(builddir, output_folder_name, test_suite_name, current_test)
+        if name in self.sanitizer_names:
+            self.output_log_prefix = name.lower() + ".log"
+            # add log path to sanitizer options
+            sanitizer_options_env_key = name + "_OPTIONS"
+            sanitizer_options = os.environ[sanitizer_options_env_key].partition(':')
+            assert len(sanitizer_options) > 0
+            sanitizer_options_env_val = ""
+            i = 0
+            while i < len(sanitizer_options):
+                if "log_path=" in sanitizer_options[i]: # remove the older log path
+                    i = i + 2
+                else:
+                    sanitizer_options_env_val += sanitizer_options[i]
+                    i = i + 1
+            sanitizer_options_env_val += ":log_path=" + os.path.join(self.output_log_path, self.output_log_prefix)
+            os.environ[sanitizer_options_env_key] = sanitizer_options_env_val
+        # re-create the test output folder (or create if it was never there). Same as log output, we delete everything from old tests
+        shutil.rmtree(self.output_log_path, ignore_errors=True)
```

Comment:
Why not use python3's pathlib for handling the creation and removal of paths (everywhere)?

Reply:
For creation, I agree it is better. For deletion Path.rmdir() will throw exception if directory exist and no way to flag other, which make the code longer and complicate things (yes, its safer, but this is just a test). so I think better stay with the old way.
i'll change the creation.